src/_pytest/_raises_group.py

```diff
@@ -0,0 +1,1009 @@
+        else:
+            self.match = match
+        self.check = check
+        self._fail_reason: str | None = None
+
+        # used to suppress repeated printing of `repr(self.check)`
+        self._nested: bool = False
+
+    @property
+    def fail_reason(self) -> str | None:
+        """Set after a call to :meth:`matches` to give a human-readable reason for why the match failed.
+        When used as a context manager the string will be printed as the reason for the
+        test failing."""
+        return self._fail_reason
+
+    def _check_check(
+        self: AbstractRaises[BaseExcT_1],
+        exception: BaseExcT_1,
+    ) -> bool:
+        if self.check is None:
+            return True
+
+        if self.check(exception):
+            return True
+
+        check_repr = "" if self._nested else " " + repr_callable(self.check)
+        self._fail_reason = f"check{check_repr} did not return True"
+        return False
+
+    def _check_match(self, e: BaseException) -> bool:
+        if self.match is None or re.search(
+            self.match,
+            stringified_exception := _stringify_exception(e),
+        ):
+            return True
+
+        maybe_specify_type = (
+            f" of {_exception_type_name(type(e))}"
+            if isinstance(e, BaseExceptionGroup)
+            else ""
+        )
+        self._fail_reason = (
+            f"Regex pattern {_match_pattern(self.match)!r}"
+            f" did not match {stringified_exception!r}{maybe_specify_type}"
+        )
+        if _match_pattern(self.match) == stringified_exception:
+            self._fail_reason += "\n  Did you mean to `re.escape()` the regex?"
+        return False
+
+    @abstractmethod
+    def matches(
+        self: AbstractRaises[BaseExcT_1], exc_val: BaseException
+    ) -> TypeGuard[BaseExcT_1]:
+        """Check if an exception matches the requirements of this AbstractRaises.
+        If it fails, :meth:`AbstractRaises.fail_reason` should be set.
+        """
+
+
+@final
+class RaisesExc(AbstractRaises[BaseExcT_co_default]):
+    """Helper class to be used together with RaisesGroups when you want to specify requirements on sub-exceptions.
+
+    You don't need this if you only want to specify the type, since :class:`RaisesGroup`
+    accepts ``type[BaseException]``.
+
+    The type is checked with :func:`isinstance`, and does not need to be an exact match.
+    If that is wanted you can use the ``check`` parameter.
+
+    :meth:`RaisesExc.matches` can also be used standalone to check individual exceptions.
+
+    Examples::
+
+        with RaisesGroups(RaisesExc(ValueError, match="string"))
+            ...
+        with RaisesGroups(RaisesExc(check=lambda x: x.args == (3, "hello"))):
+            ...
+        with RaisesGroups(RaisesExc(check=lambda x: type(x) is ValueError)):
+            ...
+
+    Tip: if you install ``hypothesis`` and import it in ``conftest.py`` you will get
```

Comment:
Can we elaborate on this a bit? Someone who does not understand the history behind this feature will have a hard time understanding this tip.

Probably also worth adding this under a `.. note::`?

Reply:
Gave it a go, and removed it from `RaisesExc` to avoid repetition. (it's valid for `RaisesExc` as well, but you're unlikely to care if only using `RaisesExc`).